File: Semester4/Advanced Programming/ThriftProj/C#Thrift/Lab2_Festival/Common/protocol/SimpleResponse.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Common.protocol
{
    [Serializable]
    public class SimpleResponse : IResponse
    {
        private ResponseType responseType;
        private object data;

        public SimpleResponse(ResponseType responseType, object data)
        {
            this.responseType = responseType;
            this.data = data;
        }

        public SimpleResponse(ResponseType responseType)
        {
            this.responseType = responseType;
            data = null;
        }
        
        public object GetData()
        {
            return data;
        }

        public ResponseType GetResponseType()
        {
            return this.responseType;
        }

        public override string ToString()
        {
            if (data == null)
                return "Response: {"
                + " ResponseType: " + responseType.ToString() + " }";
            return "Response: {"
                + " ResponseType: " + responseType.ToString()
                + " Data: " + data.ToString() + "}";
        }
    }
}

File: Semester4/Advanced Programming/ThriftProj/JavaThrift/Lab2_Hw/Networking/src/main/java/objProtocol/SimpleResponse.java
package objProtocol;

import java.io.Serializable;

public class SimpleResponse implements IResponse, Serializable {
    private Object data;
    private ResponseType responseType;

    public SimpleResponse(ResponseType responseType){
        this.responseType = responseType;
        this.data = null;
    }

    @Override
    public String toString() {
        return "SimpleResponse{" +
                "data=" + data +
                ", responseType=" + responseType +
                '}';
    }

    public SimpleResponse(ResponseType responseType, Object data){
        this.data = data;
        this.responseType = responseType;
    }

    @Override
    public Object getData() {
        return this.data;
    }

    @Override
    public ResponseType getResponseType() {
        return this.responseType;
    }
}
